common/lib/transport/connectionmanager.js

```diff
@@ -165,51 +158,77 @@ var ConnectionManager = (function() {
 			return;
 		}
 
-		/* set up the transport params */
-		/* first attempt the main host; no need to check for general connectivity first.
-		 * Inherit any connection state */
-		var mode = this.connectionKey ? 'resume' : (this.options.recover ? 'recover' : 'clean');
-		var transportParams = new TransportParams(this.options, null, mode, this.connectionKey, this.connectionSerial);
-		Logger.logAction(Logger.LOG_MINOR, 'ConnectionManager.chooseTransport()', 'Transport recovery mode = ' + mode + (mode == 'clean' ? '' : '; connectionKey = ' + this.connectionKey + '; connectionSerial = ' + this.connectionSerial));
-		var self = this;
+		function decideMode(modeCb) {
+			if(self.connectionKey) {
+				modeCb('resume');
+				return;
+			}
 
-		/* if there are no http transports, just choose from the available transports,
-		 * falling back to the first host only;
-		 * NOTE: this behaviour will never apply with a default configuration. */
-		if(!this.httpTransports.length) {
-			transportParams.host = this.httpHosts[0];
-			Logger.logAction(Logger.LOG_MINOR, 'ConnectionManager.chooseTransport()', 'No http transports available; ignoring fallback hosts');
-			this.chooseTransportForHost(transportParams, self.transports.slice(), callback);
-			return;
+			if(typeof self.options.recover === 'string') {
+				modeCb('recover');
+				return;
+			}
+
+			var recoverFn = self.options.recover,
+				lastSessionData = getFromSession && getFromSession(sessionRecoveryName);
+			if(lastSessionData && lastSessionData.host === window.location.host && typeof(recoverFn) === 'function') {
```

Comment:
Why do we need to check `lastSessionData.host == window.location.host`, I would have thought the security model in the browser would look after that for us?